### PR TITLE
Added configurable request timeout to API v5 client

### PR DESCRIPTION
**Pull request by Opelord**
Issue: KNTK-329

### CODE REVIEW

File: kentikapi/internal/api_connection/rest_client.go
Comment (mmac-m3a) [inline kentikapi/internal/api_connection/rest_client.go:164]: Shouldn't there really be a period at the end of the sentence? Doesn't one of the linters insist on that?

**Review by mmac-m3a — DISMISSED** [review]
LGTM

**Comment by mmac-m3a** [comment]
I'll wait for @Fenthick to approve too before merging.

File: kentikapi/users_integration_test.go
Comment (Fenthick) [inline kentikapi/users_integration_test.go:282]: "context deadline exceeded" string is an implementation detail of the _context_ package, so we should avoid relying on that, even in tests.
Please use something like `assert.True(t, errors.Is(err, context.DeadlineExceeded))`.

Using errors.As() ensures that test will not break when our implementation wraps the error.
Using context.DeadlineExceeded ensures that test will not break when error message changes.

See: https://go.dev/blog/go1.13-errors

File: kentikapi/users_integration_test.go
Comment (Fenthick) [inline kentikapi/users_integration_test.go:240]:
```suggestion
			name: "timeout during first request",
```
First request won't even finish, so "after" is not precise

File: kentikapi/users_integration_test.go
Comment (Fenthick) [inline kentikapi/users_integration_test.go:210]:
```suggestion
			name: "default timeout is larger than 30 ms",
```

File: kentikapi/users_integration_test.go
Comment (Fenthick) [inline kentikapi/users_integration_test.go:220]: I don't see additional value in this test case when comparing to _TestClient_GetUser.status bad request_. Please remove that test case.

File: kentikapi/users_integration_test.go
Comment (Fenthick) [inline kentikapi/users_integration_test.go:198]: It think that this test could be easily merged with TestClient_GetUser.

1. Provide serverHandlingDelay duration as test parameter and pass it to test HTTP handler.
2. Remove expectedReqs and just skip checking number of requests if expectedTimeout is true.

Please merge the tests or explain why they are separate.

**Review by Fenthick — CHANGES_REQUESTED** [review]
Good job overall (nice tests!), but we need to reduce duplication in tests.

File: kentikapi/internal/testutil/http.go
Comment (Fenthick) [inline kentikapi/internal/testutil/http.go:63]:
```suggestion
	// handlingDelay specifies the delay applied while handling the request.
```

File: kentikapi/internal/testutil/http.go
Comment (Fenthick) [inline kentikapi/internal/testutil/http.go:92]: Nit: I'd prefer that line above starting forming the response (i.e. above `rw.Header().Set("Content-Type", "application/json")`), not in the middle of it, to increase the cohesion of the block of code. Possible with a line break between them.

File: kentikapi/users_integration_test.go
Comment (Fenthick) [inline kentikapi/users_integration_test.go:203]:
```suggestion
```
Please just remove remove that comment. The second sentence is false for multiple test cases. And the first one doesn't explain anything (at least for me)

**Review by Fenthick — CHANGES_REQUESTED** [review]

File: kentikapi/internal/api_connection/rest_client.go
Comment (Fenthick) [reply kentikapi/internal/api_connection/rest_client.go]: Done

File: apiv6/kentikapi/httputil/httputil_test.go
Comment (Fenthick) [reply apiv6/kentikapi/httputil/httputil_test.go]: Done

File: kentikapi/internal/api_connection/rest_client.go
Comment (Fenthick) [reply kentikapi/internal/api_connection/rest_client.go]: Done

File: kentikapi/internal/api_connection/rest_client.go
Comment (Fenthick) [reply kentikapi/internal/api_connection/rest_client.go]: Done

File: kentikapi/internal/api_connection/rest_client.go
Comment (Fenthick) [reply kentikapi/internal/api_connection/rest_client.go]: Done

File: kentikapi/internal/api_connection/rest_client.go
Comment (Fenthick) [reply kentikapi/internal/api_connection/rest_client.go]: Done

File: kentikapi/internal/api_connection/rest_client.go
Comment (Fenthick) [reply kentikapi/internal/api_connection/rest_client.go]: Done

File: kentikapi/internal/api_connection/rest_client.go
Comment (Fenthick) [reply kentikapi/internal/api_connection/rest_client.go]: Done

File: kentikapi/internal/testutil/http.go
Comment (Fenthick) [reply kentikapi/internal/testutil/http.go]: Done

File: kentikapi/internal/testutil/http.go
Comment (Fenthick) [reply kentikapi/internal/testutil/http.go]: Done

File: kentikapi/users_integration_test.go
Comment (Fenthick) [reply kentikapi/users_integration_test.go]: Done

File: kentikapi/users_integration_test.go
Comment (Fenthick) [reply kentikapi/users_integration_test.go]: Done

File: kentikapi/users_integration_test.go
Comment (Fenthick) [reply kentikapi/users_integration_test.go]: Done

File: kentikapi/users_integration_test.go
Comment (Fenthick) [reply kentikapi/users_integration_test.go]: Done

File: kentikapi/users_integration_test.go
Comment (Fenthick) [reply kentikapi/users_integration_test.go]: Done

File: kentikapi/users_integration_test.go
Comment (Fenthick) [reply kentikapi/users_integration_test.go]: Done

File: kentikapi/users_integration_test.go
Comment (Fenthick) [reply kentikapi/users_integration_test.go]: Please apply suggestion number 2.

File: kentikapi/internal/testutil/http.go
Comment (Fenthick) [reply kentikapi/internal/testutil/http.go]: Done

File: kentikapi/internal/testutil/http.go
Comment (Fenthick) [reply kentikapi/internal/testutil/http.go]: Done

File: kentikapi/users_integration_test.go
Comment (Fenthick) [reply kentikapi/users_integration_test.go]: Done

**Review by Fenthick — CHANGES_REQUESTED** [review]
One left to go: https://github.com/kentik/community_sdk_golang/pull/54#discussion_r716675962

File: kentikapi/users_integration_test.go
Comment (Opelord) [reply kentikapi/users_integration_test.go]: Sorry, I misread. On it.

**Review by Fenthick — APPROVED** [review]

**Review by mmac-m3a — APPROVED** [review]
LGTM

**Pull request merged**
